src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java

```diff
@@ -223,34 +234,22 @@ public static void main(String[] args) throws Exception {
 
     private static void performUpload(OkHttpClient client, Map<String, Set<File>> formatToFiles, Input input)
             throws IOException {
-        String sessionId = openSession(client, input);
+        String sessionId = openSession(client, input, formatToFiles.keySet());
         for (String format : formatToFiles.keySet()) {
             Set<File> filesFormFormat = formatToFiles.get(format);
             sendRequestForFormat(client, input, format, filesFormFormat, sessionId);
         }
         closeSession(client, input, sessionId);
     }
 
-    private static String openSession(OkHttpClient client, Input input) throws IOException {
+    private static String openSession(OkHttpClient client, Input input, Collection<String> formats) throws IOException {
         HttpUrl.Builder builder = input.url.newBuilder()
                 .addPathSegments("api/projects").addPathSegment(input.project)
                 .addPathSegments("external-analysis/session")
                 .addQueryParameter("partition", input.partition);
 
-        // We track revision or branch:timestamp for the session as it should be the same for all uploads
-        if (input.commit != null) {
-            builder.addQueryParameter("revision", input.commit);
-        } else if (input.timestamp != null) {
-            builder.addQueryParameter("t", input.timestamp);
-        } else if (input.autoDetectedCommit) {
-            String commit = detectCommit();
-            if (commit == null) {
-                fail("Failed to automatically detect the commit. Please specify it manually via --commit or --timestamp");
-            }
-            builder.addQueryParameter("revision", commit);
-        } else {
-            builder.addQueryParameter("t", "HEAD");
-        }
+        String revision = handleRevisionAndBranchTimestamp(input, builder);
+        builder.addQueryParameter("message", MessageUtils.createDefaultMessage(revision, input.partition, formats));
```

Comment:
We always set the default message now, even if the user has specified a custom message, right? If the user has set a custom message, we should use that instead. 

Or we simply assume our message is always better anyways and leave it like this 😄

Reply:
of course it is ;-) 
good spot, I forgot that